src/Spago/FetchPackage.hs

```diff
@@ -185,3 +185,9 @@ getLocalCacheDir (packageName, Package{ repo = Remote _, ..}) = do
 getLocalCacheDir (_, Package{ repo = Local path }) =
   Text.unpack path
 
+
+-- | Returns the name of the cache dir based on version from `PackageName`
+--   If the branch version you are trying to install has `/`'s in the branch name replace
+--   the `/`'s and return the folder name
+getCacheVersionDir :: Text -> Text
+getCacheVersionDir = Text.replace "/" "-"
```

Comment:
I think ideally this function should be injective, i.e. different inputs map to different outputs. Admittedly a bit of an edge case, but happens with this code if you have a repo with two branches called `harry/fix-1` and `harry-fix-1`?

Reply:
:man_facepalming: Yes good point thanks!